src/cider/exec/nextgen/context/CiderSet.h

```diff
@@ -167,6 +167,78 @@ class CiderDoubleSet : public CiderSet {
   robin_hood::unordered_set<double> set_;
 };
 
+class CiderStringSet : public CiderSet {
+ public:
+  CiderStringSet() : CiderSet() {}
+
+  void insert(int8_t key_val) {
```

Comment:
I think we can impl these `insert` methods in `CiderSet`. No need to make it pure virtual function.